### ISSUE TITLE
SLF001 and type annotations

### ISSUE THREAD

**Issue by ofek** [astral-sh/ruff#9013]
Should this trigger for type annotations?

```python
from __future__ import annotations

import argparse
from typing import Any

def build_command(subparsers: argparse._SubParsersAction, defaults: Any) -> None:
    ...
```

**Comment by charliermarsh** [comment]
That is an interesting question... My initial reaction was "Yes, it should trigger", since it still represents an access on a private API. But I can see why this would be necessary in the ordinary course of business (calling public methods that return opaque private types). So perhaps it should be allowed.

(I kind of feel like such objects should be public, with private interfaces as in, e.g., https://bugs.python.org/issue41592, but recognize that's not within the client's control.)

**Comment by ofek** [comment]
Yeah your assessment is exactly correct, public APIs often do not have perfect encapsulation and expose private details.

**Comment by ofek** [comment]
Thank you very much!
